backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/PreInlineLowering.kt

```diff
@@ -29,19 +37,36 @@ internal class PreInlineLowering(val context: Context) : BodyLoweringPass {
     private val enableAssertions = context.config.configuration.getBoolean(KonanConfigKeys.ENABLE_ASSERTIONS)
 
     override fun lower(irBody: IrBody, container: IrDeclaration) {
-        irBody.transformChildrenVoid(object : IrBuildingTransformer(context) {
+        var symbolOwner = container
+        while (symbolOwner !is IrSymbolOwner)
```

Comment:
Looks somewhat cryptic. Could you provide an example for this case, in a comment?

Reply:
Well, technically, `IrDeclaration` doesn't inherit `IsSymbolOwner` but in reality look like it does. I'll remove the loop then.

Reply:
One more thing to consider here: `container` is not an arbitrary `IrDeclaration`, but it is an `IrDeclaration` with a body.
So it should be enough to make sure that all declarations with bodies are symbol owners.